### ISSUE TITLE
iOS 8 compatibilty?

### ISSUE THREAD

**Issue by hacolyte** [omorandi/TiSMSDialog#15]
**Titanium 3.4.0**
**iOS 8.0.2**
**iPhone 5S**

Hey Olivier,
I love this plugin - it worked great on iOS 7 but am having some trouble now that I have upgraded to 8. I have included console output below to assist in troubleshooting.

```
[DEBUG] [iphone, 8.0.2, 192.168.1.6] Native module:com.omorandi
[INFO]  [object ComOmorandiModule] loaded
[DEBUG] Loading: /private/var/mobile/Containers/Bundle/Application/BD73D6FB-7B1B-426D-9C4D-9FDBACCD779C/myappname.app/com.omorandi.js, Resource: com_omorandi_js
[DEBUG] SMSDialog - Listener Added - type: complete, count: 1
[INFO] [iphone, 8.0.2, 192.168.1.6] module is => [object ComOmorandiModule]
[DEBUG] <CKSMSComposeRemoteViewController: 0x18e22000> timed out waiting for fence barrier from com.apple.mobilesms.compose
[DEBUG] CoreAnimation: failed to receive fence reply: 10004003
[INFO] [iphone, 8.0.2, 192.168.1.6] Result: Operation cancelled
```

Thanks

**Comment by omorandi** [comment]
Hi, can you share the code you use for setting the recipients and showing up the sms dialog? Thanks

**Comment by rgaushell** [comment]
I'm getting this error with SMS too, but am getting a similar error just using Titanium's standard email dialog:
[INFO] :   2014-10-17 08:59:31.105 OrderBuddy[13422:23667740] <MFMailComposeRemoteViewController: 0x793cba00> timed out waiting for fence barrier from com.apple.MailCompositionService

Code to reproduce:
var emailDialog = Ti.UI.createEmailDialog();
                emailDialog.subject = "Great special";
                emailDialog.toRecipients = [];
                emailDialog.messageBody = "Hi there";
                emailDialog.open();

Maybe a bug in Titanium SDK 3.4 with iOS 8?  
Hope this helps, I'm happy to test a beta version.

**Comment by rgaushell** [comment]
hmm... stackoverflow is saying that this is a simulator-only issue (http://stackoverflow.com/questions/25895634/email-composure-ios-8).  I'll test both on a device and see what happens.

**Comment by hacolyte** [comment]
@omorandi @bittome thanks so much for the replies. I created a much smaller test case than what is implemented in my app, isolating only SMS with this module using the example provided and it built to my iPhone 5s running 8.0.2 just fine. I can confirm it does not build to the simulator running 8, but it sounds like that is to be expected based on @bittome 's comment. I went back to the code in my app, and realized at some point (probably around 3am) I put the recipients value in an array, not realizing it was already being returned as an array from the Collection.

```
phone = args.data.properties.phone;
...
smsDialog.recipients = [phone];  // returns nested array [["1231234567"]] error

smsDialog.recipients = phone;  // returns ["1231234567"] works
```

So, big syntactic 'DUH' on my part. FWIW, I ran a pared down test case using EmailDialog as well and can confirm same functionality - works on device but not simulator.
